internal/rbd/controllerserver.go

```diff
@@ -274,7 +271,16 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 				return nil, err
 			}
 		}
-		return buildCreateVolumeResponse(ctx, req, rbdVol)
+
+		if parentVol != nil && parentVol.isEncrypted() {
```

Comment:
the same need to be handled for snapshot also?

Reply:
no, that does not seem to be required

Reply:
what if during creating a clone from snapshot operation the rbd image got created but the encryption information is not stored (maybe in midway the controller restarted?)

Reply:
A next call would return a `VolumeOperationAlreadyExists` error, just like in other places where the provisioning is partially done and aborted. I do not think it is easy to recover from that automatically.

Reply:
I still feel like this is also needed for a snapshot. during vol Exist check for restoring rbd image from snapshot the parentVol is empty we need to check  if the volume encrypted if is restored from a snapshot

Reply:
```go
// check after vol exists from snapshot
if rbdSnap.isEncrypted() {
			rbdSnap.conn = rbdVol.conn.Copy()
			err = rbdSnap.copyEncryptionConfig(&rbdVol.rbdImage)
			if err != nil {
				return nil, fmt.Errorf("failed to clone encryption config: %w", err)
			}
		}
```

Reply:
I think this is now addressed with the added `repairEncryptionConfig()` function.